virttest/qemu_vm.py

```diff
@@ -3288,17 +3273,15 @@ def create(self, name=None, params=None, root_dir=None,
                                  % nic.nic_name)
                     # Update the fd and vhostfd for nic devices
                     if self.devices is not None:
-                        for device in self.devices:
-                            cmd = device.cmdline()
-                            if cmd is not None and "fd=" in cmd:
-                                new_cmd = ""
-                                for opt in cmd.split(","):
-                                    if re.match('fd=', opt):
-                                        opt = 'fd=%s' % nic.tapfds
-                                    if re.match('vhostfd=', opt):
-                                        opt = 'vhostfd=%s' % nic.vhostfds
-                                    new_cmd += "%s," % opt
-                                device._cmdline = new_cmd.rstrip(",")
+                        dev = self.devices.get_by_params({'id': nic.netdev_id})[0]
+                        net_params = {'vhostfd': nic.vhostfds.split(":")[0],
+                                      'vhostfds': nic.vhostfds,
+                                      'fd': nic.tapfds.split(":")[0],
+                                      'fds': nic.tapfds,
+                                      }
+                        for k, v in net_params.items():
+                            if k in dev.params:
+                                dev.set_param(k, v)
```

Comment:
Hello @yanan-fu, this is the root cause of the failure you mentioned. I forgot to update this part last time.

Actually `Update the fd and vhostfd for nic devices` is more like a workaround in this place, but I don't want to fix it this time, let's move this to my next patch, decoupling qemu life cycle.

Reply:
Regression test from my side all PASS with the latest code, thanks!